conan/tools/microsoft/nmakedeps.py

```diff
@@ -50,10 +50,10 @@ def format_define(define):
                     # CL env-var can't accept '=' sign in /D option, it can be replaced by '#' sign:
                     # https://learn.microsoft.com/en-us/cpp/build/reference/cl-environment-variables
                     macro, value = define.split("=", 1)
-                    if value and not value.isnumeric():
```

Comment:
Aren't we missing something here? Maybe it was not covered by tests? If value is a number, process is different?